src/browser.ts

```diff
@@ -88,33 +95,118 @@ function hasLegacyDestinations(settings: LegacySettings): boolean {
   )
 }
 
-async function flushBuffered(analytics: Analytics): Promise<void> {
+/**
+ * The names of any Analytics instance methods that can be called pre-initialization.
+ * These methods should exist statically on AnalyticsBrowser.
+ */
+type PreInitMethodName =
+  | 'trackSubmit'
+  | 'trackClick'
+  | 'trackLink'
+  | 'trackForm'
+  | 'pageview'
+  | 'identify'
+  | 'reset'
+  | 'group'
+  | 'track'
+  | 'ready'
+  | 'alias'
+  | 'debug'
+  | 'page'
+  | 'once'
+  | 'off'
+  | 'on'
+  | 'addSourceMiddleware'
+  | 'addIntegrationMiddleware'
+  | 'setAnonymousId'
+  | 'addDestinationMiddleware'
+
+/**
+ *  Represents a buffered method call that occurred before initialization.
+ */
+interface PreInitMethodCall<T extends PreInitMethodName = PreInitMethodName> {
```

Comment:
I'm unfamiliar with the section that follows the `extends` keyword here. What's happening there?

Reply:
Sure; it assigns the default type parameter of PreInitMethodName, so PreInitMethodCall can be called without an argument (if desired). 

Created a playground

https://www.typescriptlang.org/play?#code/C4TwDgpgBAaghgGwK7QLxQOQDMD2ONQA+mARnAE4YBQVokUAYgDwAqUEAHsBAHYAmAZ1iIUUdPGQQAfGKgBvKlCVQAbiIgAuKC0XKeSALZaAjFQC+NKgHoAVDdrhoAQVkLlq9VoBEuHF4Dcukr6RlDGgRZQNlYO9C7ozNh4GDI0tvZ00MxsnNz8QhKi4uoy6G7KapJaLIHuISYRVNGWAMY4PALAUADmEMAMeLKs7Fy8gsKSUgAUFN3VAJTVYjLlSuR9SOQ8ULPmre2dUCSyvf14U0n4OwXq8zRtHV1Ysj19AzgXvhh3QA

Reply:
BTW, I just pushed a commit that simplified the types a bit!

Reply:
Oh I didn't know you could do default values in generics, neat!